Accept colon(s) in YAML key names

gdubicki opened the pull request:
as it's ok according to the YAML specs and is widely used in the real world, f.e. in Puppet's Hiera (https://puppet.com/docs/puppet/7/hiera_quick.html#values_common_data)

birkenfeld commented:
This also accepts question marks, and it allows leading colons; is that intended?

gdubicki commented:
Thanks for a super-fast response @birkenfeld! 😊

The former was not intended, I fixed it now.

The latter was intended as it's the same as having a colon at any other place in the key name. I added a test for that.

birkenfeld commented:
Looks good, thanks!
